raphtory/src/algorithms/components/in_components.rs

```diff
@@ -91,13 +96,85 @@ where
     );
     AlgorithmResult::new(graph.clone(), "In Components", results_type, res)
 }
+/// Computes the in-component of a given node in the graph
+///
+/// # Arguments
+///
+/// * `node` - The node whose in-component we wish to calculate
+///
+/// Returns:
+///
+/// A Vec containing the Nodes within the given nodes in-component
+///
+pub fn in_component<'graph, G: GraphViewOps<'graph>>(node: NodeView<G>) -> Vec<NodeView<G>> {
+    let mut in_components = HashSet::new();
+    let mut to_check_stack = Vec::new();
+    node.in_neighbours().iter().for_each(|node| {
+        let id = node.node;
+        in_components.insert(id);
+        to_check_stack.push(id);
+    });
+    while let Some(neighbour_id) = to_check_stack.pop() {
+        if let Some(neighbour) = &node.graph.node(neighbour_id) {
+            neighbour.in_neighbours().iter().for_each(|node| {
+                let id = node.node;
+                if !in_components.contains(&id) {
+                    in_components.insert(id);
+                    to_check_stack.push(id);
+                }
+            });
+        }
+    }
+    in_components
+        .iter()
+        .filter_map(|vid| node.graph.node(*vid))
+        .collect()
+}
 
 #[cfg(test)]
 mod components_test {
     use super::*;
     use crate::{db::api::mutation::AdditionOps, prelude::*, test_storage};
     use std::collections::HashMap;
 
+    #[test]
+    fn in_component_test() {
+        let graph = Graph::new();
+        let edges = vec![
+            (1, 1, 2),
+            (1, 1, 3),
+            (1, 2, 4),
+            (1, 2, 5),
+            (1, 5, 4),
+            (1, 4, 6),
+            (1, 4, 7),
+            (1, 5, 8),
+        ];
+
+        for (ts, src, dst) in edges {
+            graph.add_edge(ts, src, dst, NO_PROPS, None).unwrap();
+        }
+
+        fn check_node(graph: Graph, node_id: u64, mut correct: Vec<u64>) {
```

Comment:
Not that it matters much here but as a general point, could take graph by reference, no real reason to clone.